packages/api/src/client/internal/generated/openapi/components/ActionParameterType.ts

```diff
@@ -0,0 +1,267 @@
+import { ActionParameterArrayType } from "./ActionParameterArrayType";
+import { AttachmentType } from "./AttachmentType";
+import { BooleanType } from "./BooleanType";
+import { DateType } from "./DateType";
+import { DoubleType } from "./DoubleType";
+import { IntegerType } from "./IntegerType";
+import { LongType } from "./LongType";
+import { OntologyObjectSetType } from "./OntologyObjectSetType";
+import { OntologyObjectType } from "./OntologyObjectType";
+import { StringType } from "./StringType";
+import { TimestampType } from "./TimestampType";
+
+export interface ActionParameterType_Array extends ActionParameterArrayType {
```

Comment:
What do we get by separating the `type` out?

Reply:
For openapi type unions we use `type` as the discriminator and that doesn't exist on the base components so if we have two components for example `A = { args: string[]}` and `B = {arg: strings}` but we want a union of the two for some request, it would require it to be like `{ type: "A", args:string[] } | {type: "B": arg: string}` so they're kind of like different interfaces

Reply:
Yeah im more curious why we split out the underlying data from the interface that has `type: "foo"`

Reply:
Depending on what A and B are they can be used elsewhere so if we had the data in the same interface that has `type: "foo"` we'd need to generate some duplicate code (unless I'm misunderstanding)